proposals/list-patterns-enumerables.md

```diff
@@ -11,214 +11,53 @@ The pattern syntax is unchanged:
 ```antlr
 primary_pattern
   : list_pattern
-  | length_pattern
   | slice_pattern
   | // all of the pattern forms previously defined
   ;
 ```
 
 ### Pattern compatibility
 
-A *length_pattern* is now also compatible with any type that is not *countable* but is *enumerable* — it can be used in `foreach`.
+When the type is not *indexable* or *countable*, pattern compatibility is defined as follows:
 
-A *list_pattern* is now also compatible with any type that is not *indexable* but is *enumerable*.
+- A *list_pattern* is compatible with any type that is *enumerable* — it can be used in `foreach`.
+- A *slice_pattern* is compatible with any type that is compatible with a *list_pattern* but no subpatterns are allowed and it can only appear as the last item in the list.
```

Comment:
>and it can only appear as the last item in the list. [](http://example.com/codeflow?start=120&length=52)

I don't think we should have this limitation. 
Let's add a note to open questions below. I'll queue this for LDM since didn't get a strong answer from Mads in email.